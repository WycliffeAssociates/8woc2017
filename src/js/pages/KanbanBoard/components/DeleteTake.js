import React from 'react';
import styled from 'styled-components';
import {DropTarget} from 'react-dnd';

class DeleteTake extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {

    const {connectDropTarget, isOver} = this.props;

    return connectDropTarget(
      <div style={styles.container}>
        <DeleteContainer isOver={isOver}>
          <DeleteTarget isOver={isOver}>
            <i className="material-icons">delete</i>
          </DeleteTarget>
        </DeleteContainer>
      </div>
    );
  }

}

const styles = {
  container: {
    position: 'fixed',
    bottom: '7.5vh',
    right: '18vw',
    zIndex: '98',
    width: '20.5vw',
    height: '25.5vh',
  },

};

const DeleteContainer = styled.div`
  margin: auto;
  text-align: center;
  margin-left: 9vw;
  margin-top: 7vw;
  padding: 0vw 0vw;
  `;

const DeleteTarget = styled.button`
  color: white;
  border: none;
  height: ${props => props.isOver? '100px': '75px'};
  width: ${props => props.isOver? '100px': '75px'};
  border-radius: ${props => props.isOver? '50px': '40px'};
  background: ${props => props.isOver? 'linear-gradient(to top, #820C00, #E74C3C)': 'linear-gradient(to top, #0076FF , #00C5FF )'};
  text-align: center;
  i{
    font-size: 36px;
  }
  box-shadow: 2px 4px 6px rgba(0,0,0,0.5);
`;



const deleteTarget = {
  drop(props, monitor, component) {
    const { listId } = props;
    return { listId: listId };
  },
};


export default DropTarget('TakeCard', deleteTarget, (connect, monitor) => ({
  connectDropTarget: connect.dropTarget(),
  isOver: monitor.isOver(),
  canDrop: monitor.canDrop(),
}))(DeleteTake);
