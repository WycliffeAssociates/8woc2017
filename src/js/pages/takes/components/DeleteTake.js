/**
 * Created by DennisMarchuk on 7/12/2017.
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import {Button} from "semantic-ui-react";
import { Icon } from 'semantic-ui-react'
import config from 'config/config';
import axios from 'axios'

export class DeleteTake extends Component{

        render(){
            return (
                <Button icon color="red" onClick={this.props.onDeleteTake}>
                    <Icon name='trash' size="large"/>
                </Button>
            );
        }
}

DeleteTake.propTypes = {
    onDeleteTake: PropTypes.func.isRequired
};

export default DeleteTake