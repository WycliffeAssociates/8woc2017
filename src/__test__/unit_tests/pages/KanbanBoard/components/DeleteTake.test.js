/* global it expect describe*/
import React from 'react';
import DeleteTake from '../../../../../js/pages/KanbanBoard/components/DeleteTake';
import renderer from 'react-test-renderer';



const OriginalDeleteTake = DeleteTake.DecoratedComponent;
const identity = el => el;

describe('DeleteTake DropTarget', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<OriginalDeleteTake connectDropTarget={identity} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
