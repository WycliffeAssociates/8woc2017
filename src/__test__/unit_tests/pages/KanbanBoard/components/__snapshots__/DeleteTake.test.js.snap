// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteTake DropTarget renders correctly 1`] = `
<div
  style={
    Object {
      "bottom": "7.5vh",
      "height": "25.5vh",
      "position": "fixed",
      "right": "18vw",
      "width": "20.5vw",
      "zIndex": "101",
    }
  }
>
  <div
    className="sc-bdVaJa iIHwQN"
  >
    <button
      className="sc-bwzfXH cPWCet"
    >
      <i
        className="material-icons"
      >
        delete
      </i>
    </button>
  </div>
</div>
`;
